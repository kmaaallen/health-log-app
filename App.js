import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
//Redux
import { Provider } from 'react-redux';
import { store, persistor } from './src/redux/store';
import { PersistGate } from 'redux-persist/integration/react';
//Navigation
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
//Screens
import LogPage from './src/screens/LogPage';

const Stack = createStackNavigator();

export default function App() {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <NavigationContainer>
          <Stack.Navigator>
            <Stack.Screen name="Log" component={LogPage} />
          </Stack.Navigator>
        </NavigationContainer>
      </PersistGate>
    </Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
