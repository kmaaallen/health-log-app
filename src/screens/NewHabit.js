import React from 'react';
import { StyleSheet, ScrollView, KeyboardAvoidingView, View } from 'react-native';
import { withTheme, Button, TextInput, HelperText, Paragraph } from 'react-native-paper';
//REDUX
import { connect } from 'react-redux';
import { createHabit } from '../redux/actions';
//FORM
import { useForm, Controller } from "react-hook-form";
//COMPONENTS
import { Picker } from "@react-native-picker/picker";

const styles = theme => StyleSheet.create({
    surface: {
        padding: theme.spacing.medium,
        margin: theme.spacing.medium,
    },
    pickerView: {
        borderWidth: 1,
        borderColor: '#ccc',
        borderRadius: 4,
        height: 100
    },
    topText: {
        textAlign: 'center',
        paddingTop: 5
    }
});

function NewHabit(props) {

    const { control, reset, handleSubmit, watch, formState: { errors }, register } = useForm({
        mode: 'onSubmit',
    });

    const watchCategoryPicker = watch("category", false);

    const createHabit = (formData) => {
        var category;
        formData.newCategory ? category = formData.newCategory : category = formData.category;
        props.createHabit(formData.title, formData.limit, category);
        props.navigation.navigate('Log');
    }

    return (
        <KeyboardAvoidingView behavior='padding'>
            <ScrollView>
                <Paragraph style={styles(props.theme).topText}>Create a new habit</Paragraph>
                <Controller
                    control={control}
                    render={({ field: { onChange, value } }) => (
                        <TextInput
                            label="Title"
                            onChangeText={value => onChange(value)}
                            value={value || ''} // to remove uncontrolled to controlled error
                            testID="title-input"
                        />
                    )}
                    name="title"
                    rules={{ required: true }}
                />
                {errors.title && <HelperText type="error" >Title is required</HelperText>}
                <Controller
                    control={control}
                    render={({ field: { onChange, value } }) => (
                        <TextInput
                            label="Daily limit"
                            keyboardType="numeric"
                            onChangeText={value => onChange(value)}
                            value={value || ''}
                            testID="limit-input"
                        />
                    )}
                    name="limit"
                    rules={{ required: true, min: 1 }}
                />
                {errors.limit && <HelperText type="error">{errors.limit.type == 'min' ? 'Limit must be greater than zero' : 'Limit is required'}</HelperText>}

                <Controller
                    control={control}
                    render={({ field: { onChange, value } }) => (
                        <View style={styles(props.theme).pickerView}>
                            <Picker
                                selectedValue={value}
                                onValueChange={value => onChange(value)}
                                itemStyle={{ fontSize: 16, height: 100 }}
                                testID="category-input"
                            >
                                <Picker.Item label="Please select a category" value="" key="empty" testID="select-category" />
                                <Picker.Item label="New category" value="New category" key="New category" testID="new-category" />
                                {props.categories.map((item) => (<Picker.Item key={item} label={item} value={item} />))}
                            </Picker>
                        </View>
                    )}
                    name="category"
                    rules={{ required: false }}
                />
                {watchCategoryPicker == 'New category' ? <Controller
                    control={control}
                    render={({ field: { onChange, value } }) => (
                        <TextInput
                            label="New Category"
                            onChangeText={value => onChange(value)}
                            value={value || ''}
                            testID="new-category"
                            placeholder="Create a new category"
                        />
                    )}
                    name="newCategory"
                    rules={{ required: false }}
                /> : null}
                <Button onPress={handleSubmit(createHabit)}>Ok</Button>
            </ScrollView>
        </KeyboardAvoidingView>
    );
}

const mapStateToProps = (state) => ({
    categories: [...new Set((Object.values(state.habits.habits)).map((item) => item.category))].filter(item => item != "" && item != undefined).sort(),
})

const mapDispatchToProps = (dispatch) => {
    return {
        createHabit: (title, limit, category) => dispatch(createHabit({ updated: (new Date()).valueOf(), title: title, limit: limit, category: category }))
    }
}

export default withTheme(connect(mapStateToProps, mapDispatchToProps)(NewHabit));