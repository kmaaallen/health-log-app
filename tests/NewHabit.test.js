import React from 'react';
import { Provider } from 'react-redux';
import { Provider as PaperProvider } from 'react-native-paper';
import { theme } from '../App';
import NewHabit from '../src/screens/NewHabit';
// Testing
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { createTestStore } from './utils';
import { NavigationContainer } from '@react-navigation/native';

describe('<NewHabit />', () => {
    let store, stockedStore;
    let initialState = {
        habits: {
            habits: {
                1: {
                    id: 1,
                    title: 'My test habit',
                    count: 0,
                    limit: 3,
                    category: 'Health',
                    frequency: 'Daily',
                    log: [{ updated: 1612235045000, info: { type: 'created' } }]
                },
            }
        }
    };

    beforeEach(() => {
        store = createTestStore();
        stockedStore = createTestStore(initialState);
    });

    it('has 1 child', () => {
        const tree = renderer.create(<Provider store={store} ><NewHabit theme={theme} /></Provider>).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Provider store={store} ><NewHabit theme={theme} /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders create new habit form title, limit, category inputs and submit button on first load', () => {
        // Paper provider component wrapper required for react-native-paper, provided at app level
        const { getByText, getByTestId, queryByTestId } = render(<Provider store={store}><PaperProvider theme={theme}><NewHabit /></PaperProvider></Provider>);
        // Expected form inputs on first load
        const title = getByTestId('title-input');
        const limit = getByTestId('limit-input');
        const category = getByTestId('category-input');
        const frequency = getByTestId('frequency-input');
        const submit = getByText('Ok');
        // Inputs to be present
        expect(title && limit && category && frequency && submit).toBeTruthy();
        // Expect category default option to be set to 'Please select a category'
        expect(category.props.items[category.props.selectedIndex].label).toBe('Please select a category');
        // Expect new category text input to be hidden at this stage
        expect(queryByTestId('new-category')).toBeFalsy();
    });

    it('renders additional new category input when category changes to New category option', async () => {
        // Paper provider component wrapper required for react-native-paper, provided at app level
        const { getByTestId, queryByTestId } = render(<Provider store={store}><PaperProvider theme={theme}><NewHabit /></PaperProvider></Provider>);
        // Get category input
        const category = getByTestId('category-input');
        // Change category input to 'New category'
        fireEvent(category, 'onValueChange', 'New category');
        // Expect category picker to be "New category" && expecting new category text input to be visible
        expect(category.props.items[category.props.selectedIndex].label).toBe('New category');
        await waitFor(() => queryByTestId('new-category'));
        expect(getByTestId('new-category')).toBeTruthy();
    });

    it('allows users to select from an existing category', () => {
        const nav = jest.fn();
        // Paper provider component wrapper required for react-native-paper, provided at app level
        const { getByTestId, queryByTestId, getByText } = render(<Provider store={stockedStore}><PaperProvider theme={theme}><NewHabit navigation={{ navigate: nav }} /></PaperProvider></Provider>);
        // Expecting category input to contain existing Health option
        expect(getByTestId('category-input').props.items.length).toBe(3);
        expect((queryByTestId('category-input').props.items[2].label)).toBe('Health');
    });

    it('allows users to submit a new habit', async () => {
        const nav = jest.fn();
        // Paper provider component wrapper required for react-native-paper, provided at app level
        const { getByTestId, queryByTestId, getByText } = render(<Provider store={store}><PaperProvider theme={theme}><NewHabit navigation={{ navigate: nav }} /></PaperProvider></Provider>);
        // Expecting category input to not contain any non-default options as empty store
        expect(getByTestId('category-input').props.items.length).toBe(2);
        // Submit new habit under new category
        const title = getByTestId('title-input');
        const limit = getByTestId('limit-input');
        const category = getByTestId('category-input');
        const frequency = getByTestId('frequency-input');
        const submit = getByText('Ok');
        fireEvent.changeText(title, 'My new habit');
        fireEvent.changeText(limit, '2');
        fireEvent(frequency, 'onValueChange', 'Daily');
        fireEvent(category, 'onValueChange', 'New category');
        await waitFor(() => queryByTestId('new-category'));
        fireEvent.changeText(getByTestId('new-category'), 'Fitness');
        await waitFor(() => {
            fireEvent.press(submit);
        });
        // Expect store to be updated with new habit that has new category
        expect(Object.keys(store.getState().habits.habits).length).toBe(1);
        expect(store.getState().habits.habits[1].title).toBe('My new habit');
        expect(store.getState().habits.habits[1].limit).toBe('2');
        expect(store.getState().habits.habits[1].frequency).toBe('Daily');
        expect(store.getState().habits.habits[1].category).toBe('Fitness');
    });

    it('validates errors correctly in create new habit form', async () => {
        // Paper provider component wrapper required for react-native-paper, provided at app level
        const { getByText, getByTestId, queryByText } = render(<Provider store={store}><PaperProvider theme={theme}><NewHabit /></PaperProvider></Provider>);
        // Expect store to be empty at this stage
        expect(store.getState().habits).toEqual({ "habits": {} });
        const limit = getByTestId('limit-input');
        const submit = getByText('Ok');
        await waitFor(() => {
            fireEvent.press(submit);
        });
        // Expect store to not be updated and error messages present
        expect(Object.keys(store.getState().habits.habits).length).toBe(0);
        expect(getByText('Title is required')).toBeTruthy();
        expect(getByText('Limit is required')).toBeTruthy();
        // change limit value to check other error message and submit form
        fireEvent.changeText(limit, '0');
        await waitFor(() => {
            fireEvent.press(submit);
        });
        // Expect store to not be updated and error messages present
        expect(Object.keys(store.getState().habits.habits).length).toBe(0);
        expect(getByText('Title is required')).toBeTruthy();
        expect(getByText('Limit must be greater than zero')).toBeTruthy();
    });

    it('returns to Log page on successful submission of form', async () => {
        const nav = jest.fn();
        const { getByTestId, getByText, queryByTestId } = render(<Provider store={store}><PaperProvider theme={theme}><NewHabit navigation={{ navigate: nav }} /></PaperProvider></Provider>);
        // Submit new habit under new category
        const title = getByTestId('title-input');
        const limit = getByTestId('limit-input');
        const frequency = getByTestId('frequency-input');
        const category = getByTestId('category-input');
        const submit = getByText('Ok');
        fireEvent.changeText(title, 'My new habit');
        fireEvent.changeText(limit, '2');
        fireEvent(frequency, 'onValueChange', 'Daily');
        fireEvent(category, 'onValueChange', 'New category');
        await waitFor(() => queryByTestId('new-category'));
        fireEvent.changeText(getByTestId('new-category'), 'New thing');
        await waitFor(() => {
            fireEvent.press(submit);
        });
        // Expect store to be updated with new habit that also has category Health
        expect(Object.keys(store.getState().habits.habits).length).toBe(1);
        expect(store.getState().habits.habits[1].title).toBe('My new habit');
        expect(store.getState().habits.habits[1].limit).toBe('2');
        expect(store.getState().habits.habits[1].frequency).toBe('Daily');
        expect(store.getState().habits.habits[1].category).toBe('New thing');
        expect(nav).toHaveBeenCalledWith('Log');
    });
});
