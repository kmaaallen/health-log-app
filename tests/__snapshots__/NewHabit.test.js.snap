// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewHabit /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Object {
                  "paddingTop": 5,
                  "textAlign": "center",
                },
              ],
            ],
          ]
        }
      >
        Create a new habit
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {},
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
              "zIndex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "minHeight": 64,
              },
            ]
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Title
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Title
            </Text>
          </View>
          <TextInput
            editable={true}
            maxFontSizeMultiplier={1.5}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                },
                Object {
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                Array [
                  Object {},
                ],
              ]
            }
            testID="title-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {},
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
              "zIndex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "minHeight": 64,
              },
            ]
          }
        >
          <View
            collapsable={false}
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "zIndex": 3,
              }
            }
          >
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Daily limit
            </Text>
            <Text
              collapsable={false}
              maxFontSizeMultiplier={1.5}
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 34,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Daily limit
            </Text>
          </View>
          <TextInput
            editable={true}
            keyboardType="numeric"
            maxFontSizeMultiplier={1.5}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                },
                Object {
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 4,
                  "paddingTop": 24,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                false,
                Array [
                  Object {},
                ],
              ]
            }
            testID="limit-input"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "borderColor": "#ccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 100,
          }
        }
      >
        <View>
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Daily",
                  "testID": "select-daily-frequency",
                  "textColor": undefined,
                  "value": "Daily",
                },
                Object {
                  "label": "Weekly",
                  "testID": "select-weekly-frequency",
                  "textColor": undefined,
                  "value": "Weekly",
                },
                Object {
                  "label": "Monthly",
                  "testID": "select-monthly-frequency",
                  "textColor": undefined,
                  "value": "Monthly",
                },
              ]
            }
            numberOfLines={1}
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "fontSize": 16,
                  "height": 100,
                },
              ]
            }
            testID="frequency-input"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "borderColor": "#ccc",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 100,
          }
        }
      >
        <View>
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Please select a category",
                  "testID": "select-category",
                  "textColor": undefined,
                  "value": "",
                },
                Object {
                  "label": "New category",
                  "testID": "new-category",
                  "textColor": undefined,
                  "value": "New category",
                },
              ]
            }
            numberOfLines={1}
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "fontSize": 16,
                  "height": 100,
                },
              ]
            }
            testID="category-input"
          />
        </View>
      </View>
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Ok
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
